### ISSUE TITLE
Changes not being saved

### ISSUE THREAD

**Issue by scondra** [surveyjs/editor#33]
Randomly when changing the name/title of a question then clicking on to something else the updates aren't saved. Doing an Undo and Redo sometimes seems to bring the changes back. Also, when dealing with multiple pages sometimes the questions and/or complete pages will disappear when making updates. This is very random and following the same process may not result in the same issue each time.

**Comment by andrewtelnov** [comment]
@scondra Does it happen when you are in design tab, I mean you do not click on JSON or 'Test Survey' tabs?
I was not able to reproduce it today. I will try tomorrow and ask @dmitrykurmanov to try it out as well.

Thank you,
Andrew

**Comment by scondra** [comment]
Andrew,

Yes it's in the design view. Very random. Happens when adding and updating
questions. Specifically when changing the name and title values of the
questions then going to a different question or page. I appreciate you
looking in to it.

**Comment by andrewtelnov** [comment]
@scondra We have tried to reproduce the issue, but had no luck :-(
However, there is a non-initiative behavior with a question.title. 
The default value for question title is a question name, where question could not be empty. In fact if the question title equals to question name then the survey library make the question title empty.
For example, you have the following code: 
```
myQuestion.name = "q1";
myQuestion.tittle = "Address";
```
After that you have decided to make the question name as 'Address':
```
myQuestion.name = "Address";
```
After that the title property will be set to empty string (null).
Could you check please, is it this problem?

I think, we have to make it more intuitive. For example, if title is empty then show the question.name value in the title in the designer. 

Thank you,
Andrew

**Comment by scondra** [comment]
Andrew,

Yes that is part of the issue. Another issue that is very similar is as
follows:

Copy the content of the question name and paste it in the title
Change the value of the name
Click on JSON Editor
Click on Survey Designer
Click on the question

Following this process the name value will have the updated value but the
title still gets cleared. I'm assuming its an order of operation. The name
and title were the same which initiated the clearing of the title even
though it was not obvious to the user, then the name was changed. The
result being the name has the new value but the title still got cleared.

Also, because of the title value gets cleared if it matches the name, the
title gets left out of the JSON. This can easily be managed in development
on our side to see if the title exists and use the name if not, but it may
be more user friendly to keep both values.

Thank you,

*Shawn Condra*
Mobile Application Developer

tiag® (The Informatics Applications Group, Inc.)
scondra@tiag.net |  727-439-3799  |  tiag.net

**Comment by andrewtelnov** [comment]
@scondra After further investigation, I have found that indeed, it was a bug. 
The actual value was not set correctly in case of title property, if the value was equal to the "name" property value. I was able to add a [unit test](https://github.com/andrewtelnov/surveyjs.editor/blob/7277722008e9915d307b60c43976e94f5de72d4d/tests/ObjectEditorTests.ts) and make it work.
I have updated the editor on the web site: http://surveyjs.org/builder/

Thank you,
Andrew
